src/components/PreSurveyForm/PreSurveyForm.tsx

```diff
@@ -81,10 +81,13 @@ const PreSurveyForm = () => {
       },
     });
 
+    const upi = auth.currentUser?.email?.split('@')?.[0];
+
     setUserProperties(analytics, {
       user_gender: values.gender,
       ethnicity: values.ethnicity,
       prev_experience: programmingExperienceLabel,
+      upi,
```

Comment:
The pre-survey form is never shown so this won't be reported. You could probably set the UPI user property on log-in based on their email address.

Reply:
I've removed the pre-survery form, and moved UPI user property to the login function.